lib/preview.js

```diff
@@ -189,29 +199,29 @@ const preview = async (
     hublet: 'na1',
     sessionToken,
     domains,
+    PORT,
+    protocol,
   }
 
   if (notify) {
     ignoreFile(notify);
   }
 
-  if (initialUpload) {
+  if (!skipUpload) {
     await initialPreviewBufferUpload(sessionInfo, filePaths);
   }
-  startPreviewWatchingService(sessionInfo);
-  const expressServer = await startLocalHttpService(sessionInfo);
+  const expressServer = await createLocalHttpServer(sessionInfo);
+  const previewWatcher = await startPreviewWatcher(sessionInfo);
 
   if (!noSsl) {
-    const { server, innerHTTPServer, innerHTTPSServer } = await startHttpsRedirectingService(expressServer, domains);
-    server.listen(3000);
-    console.log('Now listening on port 3000 with HTTPS')
+    const { server, innerHTTPServer, innerHTTPSServer } = await createHttpsRedirectingServer(expressServer, domains);
+    server.listen(PORT);
   } else {
     const httpServer = http.createServer(expressServer);
-
-    httpServer.listen(3000);
-    console.log('Now listening on port 3000 with HTTP');
+    httpServer.listen(PORT);
   }
-  startShadowDevService(sessionInfo);
+  startShadowDevServer(sessionInfo);
```

Comment:
Establishing a little better convention here (to match jsr as well), we `start` the shadowdevserver since that function also starts it listening on its port, but we `create` the other servers because those functions do not start the listening as well.
Also `server` provides `service` - so just calling them all servers.